CMakeLists.txt

```diff
@@ -23,7 +23,22 @@ target_include_directories(ev3dev PUBLIC
     $<INSTALL_INTERFACE:include>
     )
 
-target_compile_options(ev3dev PUBLIC -std=c++0x)
+if (CMAKE_VERSION VERSION_LESS "3.1")
+    # For older versions of cmake, this is how we enable C++11. However, this
+    # will also set any parent projects to use C++11, which may not be what we
+    # want (e.g. if using C++14).
+    target_compile_options(ev3dev PUBLIC -std=c++0x)
+else()
+    # CMake on Travis seems not to be setting the C++ version correctly, even
+    # when specified as below (maybe because it's an old version of gcc?). So
+    # let's set this as a private compile option for a belt-and-braces approach.
+    target_compile_options(ev3dev PRIVATE -std=c++0x)
```

Comment:
Are you sure it is safe to compile the library with c++11 and then link it with the code that was compiled with c++14? Are there no ABI incompatibilities?

Reply:
I don't think there should be (C++ is supposed to be backwards compatible with this kind of thing...) and it does work for me.

Also it's quite a common thing to do: lots of C++ libs are compiled for C++11 but you can still link against them with programs using newer versions of C++.